File: typescript/tests/orderPricing.test.ts
import { calculatePrice, getOrderLabel, getOrderPriority } from '../src/orderPricing';

describe('calculatePrice', () => {
  it('returns base price for standard orders', () => {
    expect(calculatePrice('standard', 100)).toBe(100);
  });

  it('applies 20% markup for premium orders', () => {
    expect(calculatePrice('premium', 100)).toBe(120);
  });

  it('applies 30% discount for wholesale orders', () => {
    expect(calculatePrice('wholesale', 200)).toBe(140);
  });

  it('applies 30% discount for small wholesale orders', () => {
    expect(calculatePrice('wholesale', 40)).toBe(28);
  });
});

describe('getOrderLabel', () => {
  it('returns correct label for each type', () => {
    expect(getOrderLabel('standard')).toBe('Standard Order');
    expect(getOrderLabel('premium')).toBe('Premium Order');
    expect(getOrderLabel('wholesale')).toBe('Wholesale Order');
  });
});

describe('getOrderPriority', () => {
  it('returns correct priority for each type', () => {
    expect(getOrderPriority('standard')).toBe(1);
    expect(getOrderPriority('premium')).toBe(3);
    expect(getOrderPriority('wholesale')).toBe(2);
  });
});

File: java/src/test/java/OrderPricingTest.java
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class OrderPricingTest {

    @Nested
    class CalculatePrice {
        @Test
        void shouldReturnBasePriceForStandard() {
            double price = OrderPricing.calculatePrice(OrderType.STANDARD, 100.0);
            assertEquals(100.0, price);
        }

        @Test
        void shouldApply20PercentPremiumForPremium() {
            double price = OrderPricing.calculatePrice(OrderType.PREMIUM, 100.0);
            assertEquals(120.0, price);
        }

        @Test
        void shouldApply30PercentDiscountForWholesale() {
            double price = OrderPricing.calculatePrice(OrderType.WHOLESALE, 200.0);
            assertEquals(140.0, price);
        }

        @Test
        void shouldApply30PercentDiscountForSmallWholesale() {
            double price = OrderPricing.calculatePrice(OrderType.WHOLESALE, 40.0);
            assertEquals(28.0, price);
        }
    }

    @Nested
    class GetOrderLabel {
        @Test
        void shouldReturnStandardLabelForStandard() {
            String label = OrderPricing.getOrderLabel(OrderType.STANDARD);
            assertEquals("Standard Order", label);
        }

        @Test
        void shouldReturnPremiumLabelForPremium() {
            String label = OrderPricing.getOrderLabel(OrderType.PREMIUM);
            assertEquals("Premium Order", label);
        }

        @Test
        void shouldReturnWholesaleLabelForWholesale() {
            String label = OrderPricing.getOrderLabel(OrderType.WHOLESALE);
            assertEquals("Wholesale Order", label);
        }
    }

    @Nested
    class GetOrderPriority {
        @Test
        void shouldReturnPriority1ForStandard() {
            int priority = OrderPricing.getOrderPriority(OrderType.STANDARD);
            assertEquals(1, priority);
        }

        @Test
        void shouldReturnPriority2ForWholesale() {
            int priority = OrderPricing.getOrderPriority(OrderType.WHOLESALE);
            assertEquals(2, priority);
        }

        @Test
        void shouldReturnPriority3ForPremium() {
            int priority = OrderPricing.getOrderPriority(OrderType.PREMIUM);
            assertEquals(3, priority);
        }
    }

}
